components/camel-zendesk/src/main/java/org/apache/camel/component/zendesk/ZendeskComponent.java

```diff
@@ -0,0 +1,114 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.zendesk;
+
+import java.util.Map;
+
+import org.apache.camel.CamelContext;
+import org.apache.camel.Endpoint;
+import org.apache.camel.component.zendesk.internal.ZendeskApiCollection;
+import org.apache.camel.component.zendesk.internal.ZendeskApiName;
+import org.apache.camel.component.zendesk.internal.ZendeskHelper;
+import org.apache.camel.impl.DefaultComponent;
+import org.apache.camel.spi.Metadata;
+import org.apache.camel.util.component.AbstractApiComponent;
+import org.zendesk.client.v2.Zendesk;
+
+/**
+ * The Zendesk Component.
+ */
+public class ZendeskComponent extends AbstractApiComponent<ZendeskApiName, ZendeskConfiguration, ZendeskApiCollection> {
+
+    @Metadata(label = "advanced")
+    Zendesk zendesk;
+
+    public ZendeskComponent() {
+        super(ZendeskEndpoint.class, ZendeskApiName.class, ZendeskApiCollection.getCollection());
+    }
+
+    public ZendeskComponent(CamelContext context) {
+        super(context, ZendeskEndpoint.class, ZendeskApiName.class, ZendeskApiCollection.getCollection());
+    }
+
+    @Override
+    protected ZendeskApiName getApiName(String apiNameStr) throws IllegalArgumentException {
+        return ZendeskApiName.fromValue(apiNameStr);
+    }
+
+    /**
+     * To use the shared configuration
+     */
+    @Override
+    public void setConfiguration(ZendeskConfiguration configuration) {
+        super.setConfiguration(configuration);
+    }
+
+    /**
+     * To use the shared configuration
+     */
+    @Override
+    public ZendeskConfiguration getConfiguration() {
+        return super.getConfiguration();
+    }
+
+    /**
+     * To use a shared {@link Zendesk} instance.
+     * 
+     * @return the shared Zendesk instance
+     */
+    public Zendesk getZendesk() {
+        return zendesk;
+    }
+
+    @Override
+    protected Endpoint createEndpoint(String uri, String methodName, ZendeskApiName apiName,
+            ZendeskConfiguration endpointConfiguration) {
+        endpointConfiguration.setMethodName(methodName);
+        return new ZendeskEndpoint(uri, this, apiName, methodName, endpointConfiguration);
+    }
+
+    @Override
+    protected void doStart() throws Exception {
+        super.doStart();
+
+        if (zendesk == null) {
+            if (configuration != null) {
+                zendesk = ZendeskHelper.create(configuration);
+            } else {
+                throw new IllegalArgumentException("Unable to connect, Zendesk component configuration is missing");
+            }
+        }
+    }
+
+    @Override
+    protected void doStop() throws Exception {
+        if (zendesk != null) {
```

Comment:
super.doStop is missing. Also I wonder if zendesk is closable so you can close with IOHelper.close instead?